lib/src/api/api.dart

```diff
@@ -15,8 +15,13 @@ class ArweaveApi {
   Future<http.Response> get(String endpoint) =>
       _client.get(_getEndpointUri(endpoint));
 
-  Future<http.Response> post(String endpoint, {dynamic body}) =>
-      _client.post(_getEndpointUri(endpoint), body: body);
+  Future<http.Response> post(String endpoint, {dynamic body}) {
+    try {
+      return _client.post(_getEndpointUri(endpoint), body: body);
+    } catch (e) {
+      rethrow;
```

Comment:
@jdaev did we confirm that this is strictly necessary? It really appears pointless on the face of it.

Reply:
I can try another upload with this removed

Reply:
[Removed](https://github.com/ardriveapp/arweave-dart/pull/14/commits/839deebe185f832cb6ce67be2f81294ea712481cl)